### ISSUE TITLE
Qwen3Model's MoE block discards router logits, so no load-balancing aux loss is possible

### ISSUE THREAD

**Issue by mira687** [openlanguagemodel/openlanguagemodel#82]
`Qwen3Model` routes its MoE layers through a block that discards router logits, so there is no way to compute a router load-balancing auxiliary loss for it. For a sparse MoE that is a functional gap, not just a missing accessor: load balancing is what stops the router collapsing onto a handful of experts.

Found while adding reference-parity coverage in #78, where it limits what the MoE case can honestly claim.

### Where it happens

`Qwen3Model` builds its FFN as `SwiGLUMoEFFN`, which extends `MoEFeedForwardBase`. In `src/olm/nn/feedforward/moe_base.py`:

```python
def forward(self, x: torch.Tensor) -> torch.Tensor:
    ...
    top_k_indices, top_k_weights = self.router(x)   # (batch, seq, top_k)
```

Two values are unpacked and only the output tensor is returned, so the router's logits are unreachable from outside the block.

The framework can already do this the other way — `olm.nn.moe.MoEFFN` returns `(output, router_logits)`, and `src/olm/models/alibaba/qwen3_5.py` uses it and stores `last_router_logits`. So there are two MoE implementations with different contracts, and the advertised Qwen3 preset is on the one that drops the information.

### Why it matters

Training a top-k MoE on the LM loss alone lets the router collapse onto a few experts: they get all the gradient, the rest stay near their initialisation, and effective capacity silently drops to a fraction of the parameter count. The standard remedy is an auxiliary load-balancing term, which is why `Qwen3MoeConfig` carries `router_aux_loss_coef` and `Qwen3MoeForCausalLM` can return router logits. With `Qwen3Model` there is no way to add that term without editing the library.

For reference-parity work this also caps what can be verified. #78 compares next-token cross entropy on both sides, which excludes the auxiliary objective entirely — so the two models can agree to ~1e-7 on logits and gradients and still optimise different objectives in real MoE training. That is now stated as a documented deviation on the case rather than left implied, but the underlying gap is here.

### Options

1. Point `Qwen3Model` at `olm.nn.moe.MoEFFN`, which already returns router logits. Cleanest, but changes the module tree and therefore state-dict keys, so it wants a deprecation note.
2. Have `MoEFeedForwardBase.forward` return router logits too, or stash them like `qwen3_5.py` does. Smaller, but leaves two divergent MoE paths in the tree.
3. Converge the two implementations. Best long-term, biggest change.

I have not attempted any of these — it is a library change and #78 is a test PR. Flagging it for a maintainer's call on which direction is wanted; happy to implement whichever, and to add a training-parity case covering the auxiliary loss once router logits are reachable.
